src/main/java/step4/RacingCar.java

```diff
@@ -0,0 +1,81 @@
+package step4;
+
+
+import step4.dto.InputDto;
+import step4.view.ResultView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RacingCar {
+
+    private final List<Car> carList;
+    private final InputDto inputDto;
+
+    public RacingCar(InputDto inputDto) {
+        this.carList = createCar(inputDto.getCarNameList());
+        this.inputDto = inputDto;
+    }
+
+    public void start() {
+        int tryCount = inputDto.getTryCount();
+
+        for (int i = 0; i < tryCount; ++i) {
+            move();
+        }
+
+        // 이런식으로 Main 클래스에서 start 메소드 부르고 start 에서 findWinner() 를 부르는 식으로 호출 깊이가 깊어질 수록 좋지 않겠죠 .. ?? ㅠ
+        findWinner();
+    }
+
+    private void move() {
+        for (int i = 0; i < carList.size(); ++i) {
+            Car car = carList.get(i);
+            car.carCanGoJudge(RandomNumber.randomNumberGenerator());
+
+            // ResultView.print(car);  => 이렇게 바로 호출하는거랑 아래처럼 메소드를 통해서 호출하는거랑 차이가 있을까요??
+            // 큰 차이가 없는 기분이네요 ㅠ ㅠ (메소드 끼리 괜한 의존관계를 만드는 것이 아닐지 의문입니다..)
+            print(car);
+        }
+
+        System.out.println();
+    }
+
+    private void print(Car car) {
+        ResultView.print(car);
+    }
+
+    private List<Car> createCar(List<String> carNameList) {
+        List<Car> cars = new ArrayList<>();
+
+        for (int i = 0; i < carNameList.size(); ++i) {
+            cars.add(new Car(carNameList.get(i)));
+        }
+
+        return cars;
+    }
+
+    private void findWinner() {
+        int maxDistance = maxDistance();
+
+        // lambda stream 이 아직 어색한데 이렇게 하니 확실히 엄청 편한 거 같습니다 !!
+        String winnerList = carList.stream()
+                .filter(car -> car.getDistance() == maxDistance)
```

Comment:
getter를 사용하여 Car 객체의 정보를 노출하는 방법보단 Car 객체에게 물어보면 어떨까요 :) ?
```suggestion
                .filter(car -> car.isWinner(maxDistance))
```